pkg/plugin/processor/builtin/unwrap_opencdc.go

```diff
@@ -0,0 +1,294 @@
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/goccy/go-json"
+)
+
+type unwrapOpenCDC struct {
+	sdk.UnimplementedProcessor
+
+	logger      log.CtxLogger
+	fieldRefRes sdk.ReferenceResolver
+}
+
+func newUnwrapOpenCDC(logger log.CtxLogger) sdk.Processor {
+	return &unwrapOpenCDC{logger: logger}
+}
+
+func (u *unwrapOpenCDC) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "unwrap.opencdc",
+		Summary: "A processor that unwraps the OpenCDC record saved in one of record's fields.",
+		Description: `
+The unwrap.opencdc processors unwraps the OpenCDC record saved in one of the record's fields.
+This is useful in situations where a record goes through intermediate systems before 
+being written to a final destination. In these cases, the original OpenCDC record is
+part of the payload read from the intermediate system and needs to be unwrapped before
+being written to the destination.
+'`,
+		Version: "v0.1.0",
+		Author:  "Meroxa, Inc.",
+		Parameters: config.Parameters{
+			"field": config.Parameter{
+				Default: ".Payload.After",
+				Description: `
+A reference to the field which contains the OpenCDC record.
+
+For more information about record references, see: https://github.com/ConduitIO/conduit-processor-sdk/blob/cbdc5dcb5d3109f8f13b88624c9e360076b0bcdb/util.go#L66.`,
+				Type: config.ParameterTypeString,
+			},
+		},
+	}, nil
+}
+
+func (u *unwrapOpenCDC) Configure(_ context.Context, m map[string]string) error {
+	field, ok := m["field"]
+	if !ok {
+		return cerrors.New("missing required parameter 'field'")
+	}
+
+	if !strings.HasPrefix(field, ".Payload") {
+		return cerrors.Errorf("only payload can be unwrapped, field given: %v", field)
+	}
+
+	rr, err := sdk.NewReferenceResolver(field)
+	if err != nil {
+		return cerrors.Errorf("invalid reference: %w", err)
+	}
+
+	u.fieldRefRes = rr
+	return nil
+}
+
+func (u *unwrapOpenCDC) Open(context.Context) error {
+	return nil
+}
+
+func (u *unwrapOpenCDC) Process(_ context.Context, records []opencdc.Record) []sdk.ProcessedRecord {
+	out := make([]sdk.ProcessedRecord, 0, len(records))
+	for _, rec := range records {
+		proc := u.processRecord(rec)
+		out = append(out, proc)
+		if _, ok := proc.(sdk.ErrorRecord); ok {
+			return out
+		}
+	}
+
+	return out
+}
+
+func (u *unwrapOpenCDC) processRecord(rec opencdc.Record) sdk.ProcessedRecord {
+	ref, err := u.fieldRefRes.Resolve(&rec)
+	if err != nil {
+		return sdk.ErrorRecord{Error: cerrors.Errorf("failed resolving record reference: %w", err)}
+	}
+
+	var data opencdc.StructuredData
+	switch v := ref.Get().(type) {
+	case opencdc.RawData:
+		// unmarshal raw data to structured
+		if err := json.Unmarshal(v.Bytes(), &data); err != nil {
+			return sdk.ErrorRecord{Error: cerrors.Errorf("failed to unmarshal raw data as JSON: %w", err)}
+		}
+	case opencdc.StructuredData:
+		data = v
+	case nil:
+		return sdk.ErrorRecord{Error: cerrors.New("field to unmarshal is nil")}
```

Comment:
```suggestion
		return sdk.ErrorRecord{Error: cerrors.New("field to unmarshal, nil value")}
```

Reply:
I could be missing something, but how exactly does this improve the messaging?

Reply:
I don't get it either 🤔

Reply:
lol nvmnd in my mind it looked better